packages/opentelemetry-core/src/platform/node/id.ts

```diff
@@ -14,23 +14,27 @@
  * limitations under the License.
  */
 
-import * as crypto from 'crypto';
-
 const SPAN_ID_BYTES = 8;
 const TRACE_ID_BYTES = 16;
 
 /**
  * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
  * characters corresponding to 128 bits.
  */
-export function randomTraceId(): string {
-  return crypto.randomBytes(TRACE_ID_BYTES).toString('hex');
-}
+export const randomTraceId = getIdGenerator(TRACE_ID_BYTES);
 
 /**
  * Returns a random 8-byte span ID formatted/encoded as a 16 lowercase hex
  * characters corresponding to 64 bits.
  */
-export function randomSpanId(): string {
-  return crypto.randomBytes(SPAN_ID_BYTES).toString('hex');
+export const randomSpanId = getIdGenerator(SPAN_ID_BYTES);
+
+const SHARED_BUFFER = Buffer.allocUnsafe(TRACE_ID_BYTES);
+function getIdGenerator(bytes: number): () => string {
+  return function generateId() {
+    for (let i = 0; i < bytes; i++) {
+      SHARED_BUFFER[i] = Math.floor(Math.random() * 256);
```

Comment:
Not sure if it improves performance but you could use more then 8 bits from the `Math.random()` result, e.g. `Math.floor(Math.random()* 2**32)` and then some bit operations - or use of an `Uint32Array` as storage for the `SHARED_BUFFER`.

Reply:
33e9a4a186e740f1bc61a71dc2d27fd497c74b4e

Reply:
```
  crypto                                     x   407,897 ops/sec ±1.49% (88 runs sampled)
  pseudorandom                               x 2,761,125 ops/sec ±1.14% (89 runs sampled)
  pseudorandom no alloc                      x 3,322,039 ops/sec ±0.92% (93 runs sampled)
  pseudorandom minimize random calls         x 4,853,160 ops/sec ±0.58% (94 runs sampled)
```